statsmodels/multivariate/manova.py

```diff
@@ -0,0 +1,334 @@
+    df2 = v - r + q
+    F = df2 / df1 * sigma
+    results.loc["Roy’s greatest root", 'Num DF'] = df1
+    results.loc["Roy’s greatest root", 'Den DF'] = df2
+    results.loc["Roy’s greatest root", 'F Value'] = F
+    pval = stats.f.sf(F, df1, df2)
+    results.loc["Roy’s greatest root", 'Pr > F'] = pval
+    return results.iloc[:, [4, 2, 0, 1, 3]]
+
+
+class MANOVA(Model):
+    """
+    Multivariate analysis of variance
+
+
+    Parameters
+    ----------
+    endog : array-like
+        Dependent variables (DV). A n_sample x n_y_var array where n_sample is
+        the number of observations and n_y_var is the number of DV.
+
+    exog : array-like
+        Independent variables (IV). A n_sample x n_x_var array where n is the
+        number of observations and n_x_var is the number of IV. An intercept is
+        not included by default and should be added by the user (models
+        specified using a formula include an intercept by default)
+
+    Attributes
+    -----------
+    df_resid : float
+        The number of observation `n` minus the number of IV `q`.
+
+    endog : array
+        See Parameters.
+
+    exog : array
+        See Parameters.
+
+    design_info : patsy.DesignInfo
+        Contain design info for the independent variables if model is
+        constructed using `from_formula`
+
+    """
+    def __init__(self, endog, exog, design_info=None, **kwargs):
+        Y, X = endog, exog
+        self.design_info = design_info
+        out = fit_manova(X, Y)
+        self.reg_coeffs, self.df_resid, self.inv_cov_, self.YYBXXB_ = out
+        super(MANOVA, self).__init__(endog, exog)
+
+    @classmethod
+    def from_formula(cls, formula, data, subset=None, drop_cols=None,
+                     tests=None, *args, **kwargs):
+        mod = super(MANOVA, cls).from_formula(formula, data,
+                                              subset=subset,
+                                              drop_cols=drop_cols,
+                                              tests=tests,
+                                              *args, **kwargs)
+        return mod
+
+    def test(self, H=None):
+        """
+        Testing the genernal hypothesis
+            L * B * M = 0
+        for each tuple (name, L, M) in `H` where B is the coefficient matrix
+        for the linear model Y = X * B
+
+        Parameters
+        ----------
+        H: A list of tuples
+           Hypothesis to be tested. Each element is a tuple (name, L, M)
+           containing a string `name`, the contrast matrix L and the transform
+           matrix M for transforming dependent variables, respectively. If M is
+           `None`, it would be set to an identity matrix (i.e. no dependent
+           variable transformation).
+           If `H` is None: 1) the effect of each independent variable on the
+           dependent variables will be tested. Or 2) if model is created using
+           `from_formula`,  H will be created according to `design_info`. 1)
```

Comment:
"`from_formula`" -> "a formula"

`from_formula` is hidden from users in the standard use case, and users might not know that it is used internally. (i.e. they only see the shortcut alias when using statsmodels.formula.api.